src/compile/scale.ts

```diff
@@ -65,31 +65,28 @@ export function type(fieldDef: FieldDef, channel: Channel, mark: Mark): string {
         // Also, if we support color ramp, this should be ordinal too.
         return 'time'; // time has order, so use interpolated ordinal color scale.
       }
-      if (contains([ROW, COLUMN, SHAPE], channel)) {
```

Comment:
Seems odd that we override the user preference. If they really want to use something wrong, why not?